### DISCUSSION TITLE
up-target refreshes the first <div> in body along with the provided targets

### DISCUSSION THREAD

**Discussion by ragsub** [unpoly/unpoly#444] [Q&A]
I have the below link where I would want 2 html elements with different id's to be updated.
> `<a href="some url"  id="accept" up-target="#header_table_5, #accordion_body_5" class="btn btn-sm btn-primary"><i class="bi bi-cart-check"></i> Accept</a>`

When I click on the link, the elements with 2 id's (#header_table_5 and #accordion_body_5) get updated. This is good. But in addition, the first div in my `<body>` also gets updated. The first div in my body is the bootstrap menu, so this results in the menu getting duplicated on the refreshed page.

I have temporarily solved the problem by adding an empty div `(<div></div>)` as the first div tag in body. But wanted to check if I am doing something wrong or if this is an issue.

**Comment by triskweline** [comment]
This should not happen. The only way that other elements can be targeted without being explicitly listed in `[up-target]` if that other element has an [`[up-hungry]`](https://unpoly.com/up-hungry) attribute, or is listed in `up.radio.config.hungrySelectors`.

Can you share a [Glitch project](https://glitch.com/edit/#!/empty-unpoly-project) that reproduces the issue?

**Comment by ragsub** [comment]
Thank you! the part around `up-hungry` helped me to solve the problem. I have explained it below.

My html earlier looked as below. The up-hungry was on the `<div>` wrapping around the bootstrap messages, but it did not have an id to uniquely identify it. So an additional `div` got added to the `up-target` and since there was no unique id, it was picking up the first `div` in the body which happened to be the code for the bootstrap menu.
`<body>`
 `<div class="mb-1"><<code for bootstrap menu>></div>`
 `<div class="container" id="main_body">`
 `<div up-hungry><<bootstrap messages>></div>`
 `</div>`
 `<<some other code>>`
`</body>`

I have now changed the code as given below (added id="messages" to the `div` tag of bootstrap messages) and it works as expected.

`<body>`
 `<div class="mb-1"><<code for bootstrap menu>></div>`
 `<div class="container" id="main_body">`
 `<div id="messages" up-hungry><<bootstrap messages>></div>`
 `</div>`
 `<<some other code>>`
`</body>`

The way this feature works is logical. I was stuck on why the additional unrelated div was getting added. Thank you so much for the clarification.

**Comment by triskweline** [comment]
I'm glad it helped!

The issue with auto-dervied selectors matching the wrong element is something that will be solved in Unpoly 3. See slides about strict target derivation starting [here](http://triskweline.de/unpoly3-slides/#66).
